pyrdp/mitm/X224MITM.py

```diff
@@ -99,18 +99,24 @@ def onConnectionConfirm(self, pdu: X224ConnectionConfirmPDU):
         :param _: the connection confirm PDU
         """
 
-        # X224 Response
-        protocols = NegotiationProtocols.SSL if self.originalRequest.tlsSupported else NegotiationProtocols.NONE
+        # FIXME: In case the server picks anything other than what we support, PyRDP is
+        #        likely going to be unable to complete the handshake with the server.
+        #        This should not happen since we are intercepting and spoofing the NEG_REQ,
+        #        though.
+        # protocols = NegotiationProtocols.SSL if self.originalRequest.tlsSupported else NegotiationProtocols.NONE
 
         parser = NegotiationResponseParser()
         response = parser.parse(pdu.payload)
         if isinstance(response, NegotiationFailurePDU):
             self.log.info("The server failed the negotiation. Error: %(error)s", {"error": NegotiationFailureCode.getMessage(response.failureCode)})
             payload = pdu.payload
         else:
-            payload = parser.write(NegotiationResponsePDU(NegotiationType.TYPE_RDP_NEG_RSP, 0x00, protocols))
+            # payload = parser.write(NegotiationResponsePDU(NegotiationType.TYPE_RDP_NEG_RSP, 0x00, protocols))
```

Comment:
Oops, I think we still need to set the flags to `0x00` here otherwise the server might report that it supports Remote Admin, GFX, or a few other things that might cause the client to behave in an unintended way.